siac/filecmd.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/NebulousLabs/Sia/sia/components"
+)
+
+var (
+	fileCmd = &cobra.Command{
+		Use:   "file [upload|download|status]",
+		Short: "Perform file actions",
+		Long:  "Generate a new address, send coins to another file, or view info about the file.",
+		Run:   wrap(filestatuscmd),
+	}
+
+	fileUploadCmd = &cobra.Command{
+		Use:   "file upload [file] [nickname] [pieces]",
+		Short: "Upload a file",
+		Long:  "Upload a file using a given nickname and split across 'pieces' hosts.",
+		Run:   wrap(fileuploadcmd),
+	}
+
+	fileDownloadCmd = &cobra.Command{
+		Use:   "file download [nickname] [filename]",
+		Short: "Download a file",
+		Long:  "Download a previously-uploaded file to a specified destination.",
+		Run:   wrap(filedownloadcmd),
+	}
+
+	fileStatusCmd = &cobra.Command{
+		Use:   "file status",
+		Short: "View a list of uploaded files",
+		Long:  "View a list of files that have been uploaded to the network.",
+		Run:   wrap(filestatuscmd),
+	}
+)
+
+func fileuploadcmd(file, nickname, pieces string) {
+	err := callAPI(fmt.Sprintf("/file/upload?file=%s&nickname=%s&pieces=%s", file, nickname, pieces))
```

Comment:
I think this much nesting is pretty much the maximum amount of nesting that should happen.